Adicionar validação de data para novo fator a partir de 22/02/2025

SergioVago opened the pull request:
Adicionei uma nova validação para o fator de data, aplicável a datas posteriores a 22/02/2025. A validação foi implementada como um retorno separado, em "vencimentoApos22022025", garantindo a retrocompatibilidade com o pacote original.

SergioVago committed:
feat: add new date validation after 22/02/2025

SergioVago committed:
feat: add docs

SergioVago committed:
feat: fix tests

mrmgomes commented on src/boleto-utils.d.ts:
```diff
@@ -20,12 +20,26 @@ declare module '@mrmgomes/boleto-utils' {
+     * Verifica a numeração, o tipo de código inserido e o tipo de boleto após 22/02/2025 e retorna a data de vencimento.
+     * Requer numeração completa (com ou sem formatação) e tipo de código que está sendo inserido (CODIGO_DE_BARRAS ou LINHA_DIGITAVEL).
+     */
+    function identificarDataApos22022025(codigo: string, tipoCodigo: string): Date;
```
Seria melhor dar um nome mais significativo para essa função. 
Por ex.: `identificarDataComNovoFator`. E na doc especificar a razão.

SergioVago replied:
eu tinha pensado em utilizar a ideia de NovoFator, porém vamos cair no mesmo problema daqui a 25 anos, quando tiver um novo fator.

Talvez até identificarDataComNovoFator2025 faça bastante sentido. O que você acha?

mrmgomes replied:
Funciona também.
Vai ser um problema só daqui a 25 anos. Daqui pra lá a gente pensa num nome melhor haha.

mrmgomes commented on src/boleto-utils.d.ts:
```diff
@@ -98,6 +112,7 @@ declare module '@mrmgomes/boleto-utils' {
         codigoBarras: string;
         linhaDigitavel: string;
         vencimento: string;
+        vencimentoApos22022025: string;
```
Seria melhor dar um nome mais significativo para esse atributo.
 `vencimentoComNovoFator`, por exemplo. E na doc especificar a razão.

SergioVago committed:
feat: improve docs and fix "novo fator" calc

add: new function identificarDataComNovoFator2025 on readme
change: function name from identificarDataApos22022025 to identificarDataComNovoFator2025
fix: new fact calculus. Include the 1000 subtract
fix: tests using the correct new factor calculus
